### DISCUSSION TITLE
AVX-512 implementation for SHA256 porep

### DISCUSSION THREAD

**Discussion by xinaxu** [filecoin-project/rust-fil-proofs#1575] [Ideas]
According to this benchmark https://github.com/minio/sha256-simd#performance and slack message https://filecoinproject.slack.com/archives/C0179RNEMU4/p1607522912258500?thread_ts=1607314779.484100&cid=C0179RNEMU4
Using intel AVX-512 will perform better than SHA extension by 20%-100%. Wonder if it's possible to have this implemented so more storage providers with intel hardware can be brought to the ecosystem.
The AVX-512 assembly code for SHA256 is here 
https://github.com/minio/sha256-simd/blob/master/sha256blockAvx512_amd64.asm

**Comment by parkan** [comment]
:+1: for this, our fleet is all 2nd gen Xeons that support AVX512 but not sha_ni
